SafeExamBrowser.UserInterface.Mobile/Controls/Taskbar/NetworkControl.xaml.cs

```diff
@@ -46,21 +46,52 @@ private void InitializeWirelessNetworkControl()
 
 			adapter.Changed += () => Dispatcher.InvokeAsync(Update);
 			Button.Click += (o, args) => Popup.IsOpen = !Popup.IsOpen;
-			Button.MouseLeave += (o, args) => Task.Delay(250).ContinueWith(_ => Dispatcher.Invoke(() => Popup.IsOpen = Popup.IsMouseOver));
+			var lastOpenedBySpacePress = false;
+			Button.PreviewKeyDown += (o, args) =>
+			{
+				if (args.Key == System.Windows.Input.Key.Space)                 // for some reason, the popup immediately closes again if opened by a Space Bar key event - as a mitigation, we record the space bar event and leave the popup open for at least 3 seconds
+				{
+					lastOpenedBySpacePress = true;
+				}
+			};
+			Button.MouseLeave += (o, args) => Task.Delay(250).ContinueWith(_ => Dispatcher.Invoke(() =>
+			{
+				if (Popup.IsOpen && lastOpenedBySpacePress)
+				{
+					return;
+				}
+				Popup.IsOpen = Popup.IsMouseOver;
+			}));
 			Popup.CustomPopupPlacementCallback = new CustomPopupPlacementCallback(Popup_PlacementCallback);
-			Popup.MouseLeave += (o, args) => Task.Delay(250).ContinueWith(_ => Dispatcher.Invoke(() => Popup.IsOpen = IsMouseOver));
+			Popup.MouseLeave += (o, args) => Task.Delay(250).ContinueWith(_ => Dispatcher.Invoke(() =>
+			{
+				if (Popup.IsOpen && lastOpenedBySpacePress)
+				{
+					return;
+				}
+				Popup.IsOpen = IsMouseOver;
+			}));
 			WirelessIcon.Child = GetWirelessIcon(0);
 
 			Popup.Opened += (o, args) =>
 			{
 				Background = Brushes.LightGray;
 				Button.Background = Brushes.LightGray;
+				Task.Delay(100).ContinueWith((task) => Dispatcher.Invoke(() =>
+				{
+					var btn = WirelessNetworksStackPanel.Children[0] as NetworkButton;
```

Comment:
See comment above regarding `IndexOutOfRangeException`.

Reply:
Thanks for the hint, this is now fixed.